### PR TITLE
Remove checked in config

### PR DESCRIPTION
**Pull request by tanya-borisova**
- `config.json` is now in .gitignore and is designed to be filled in on a local machine by the person conducting the experiment
- The existing `config.json` has been replaced with a `config.sample.json`
- CI now referes to it's own separate config that lives in `.github/workflows/config.json`
- Updated the README to explain this

## Not part of this PR 
- Think about having multiple configs checked in the repository for posterity and reproducibility of experiments

### CODE REVIEW

**Review by ritesh-modi — APPROVED** [review]
LGTM

File: rag_experiment_accelerator/config/config.py [hunk 1]
```diff
@@ -102,7 +102,8 @@ def __init__(
             else ChunkingStrategy.BASIC
         )
         self.AZURE_DOCUMENT_INTELLIGENCE_MODEL = config_json.get(
-            "azure_document_intelligence_model", "prebuilt-read")
+            "azure_document_intelligence_model", "prebuilt-read"
```
Comment (tanya-borisova) [inline rag_experiment_accelerator/config/config.py:105]: I'm surprised that these are here, looks like the linters are fighting again ⚔️

**Review by martinpeck — APPROVED** [review]
LGTM, but made a couple of comments. Thanks @tanya-borisova !

wine-gif-2

File: README.md [hunk 2]
```diff
@@ -163,7 +163,7 @@ az deployment group create --subscription <subscription-id> --resource-group <re
 
 To use the **RAG Experiment Accelerator**, follow these steps:
 
-1. Modify the `config.json` file with the hyperparameters for your experiment.
+1. Copy the provided `config.sample.json` file to a file named `config.json` and change any hyperparameters to tailor to your experiment.
```
[the remark this answers is not in the document]
Comment (tanya-borisova) [reply README.md:166]: I’ve added this below in the same file, next to the description of the config parameters - feels appropriate

File: rag_experiment_accelerator/config/config.py (hunk 1 above)
Comment (tanya-borisova) [reply rag_experiment_accelerator/config/config.py:105]: The problem is, linter seems happier with this version of the file, so I might have to merge that

**Pull request merged**